### PR TITLE
Wc 147/date picker a11y

### PR DESCRIPTION
**Pull request by akdetrick**
#### Related issues
Fixes https://meetup.atlassian.net/browse/WC0147

#### Description
Replaces `flatpickr` standalone usage in `CalendarComponent` with the `react-flatpickr` React component. This enables keyboard navigation and updates underlying `flatpickr` version.

### CODE REVIEW

**Comment by akdetrick** [comment]
`Prelim` until usage is verified working in `mup-web`

**Comment by akdetrick** [comment]
✅  verified this works in `mup-web`

File: src/forms/CalendarComponent.jsx [hunk 1]
```diff
@@ -3,151 +3,90 @@ import React from 'react';
...
+					'field--error': Boolean(error)
+				},
 				className
 			)
 		};
 
+		const options = {
+			altInput: true,
+			altFormat: 'D M d, Y',
+			nextArrow: `<span class="svg svg--chevron-right">
```
Comment (mperrotti) [inline src/forms/CalendarComponent.jsx]: Do we have to use markup? Can't we just return it from the `<Icon/>` component?

Comment (akdetrick) [reply src/forms/CalendarComponent.jsx]: I believe we do; these options get passed directly to `flatpickr` and we need to pass an html string

Comment (mperrotti) [reply src/forms/CalendarComponent.jsx]: Ok, I figured it was something like that

**Pull request merged**